File: face_recognition/test_param_filtering.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
param1/param2 필터링 테스트 스크립트
"""

import requests
import json
import base64
from datetime import datetime

# 테스트 서버 설정
BASE_URL = "http://localhost:5002"

def test_health_check():
    """헬스 체크"""
    print("1. 헬스 체크 테스트")
    response = requests.get(f"{BASE_URL}/api/face/health")
    print(f"상태: {response.status_code}")
    print(f"응답: {json.dumps(response.json(), indent=2, ensure_ascii=False)}")
    print("-" * 50)
    return response.json()

def load_test_image(image_path):
    """테스트 이미지 로드"""
    with open(image_path, 'rb') as f:
        image_data = base64.b64encode(f.read()).decode()
    return f"data:image/jpeg;base64,{image_data}"

def test_register_with_params(member_id, image_data, param1, param2):
    """param1, param2를 사용한 얼굴 등록 테스트"""
    print(f"\n2. 얼굴 등록 테스트 - member_id: {member_id}, param1: {param1}, param2: {param2}")
    
    data = {
        "member_id": member_id,
        "image": image_data,
        "security_level": 3,
        "notes": f"테스트 등록 - {datetime.now()}",
        "param1": param1,
        "param2": param2
    }
    
    response = requests.post(f"{BASE_URL}/api/face/register", json=data)
    print(f"상태: {response.status_code}")
    print(f"응답: {json.dumps(response.json(), indent=2, ensure_ascii=False)}")
    print("-" * 50)
    return response.json()

def test_recognize_with_filtering(image_data, param1=None, param2=None):
    """param1, param2 필터링을 사용한 얼굴 인식 테스트"""
    filter_desc = []
    if param1:
        filter_desc.append(f"param1={param1}")
    if param2:
        filter_desc.append(f"param2={param2}")
    filter_str = ", ".join(filter_desc) if filter_desc else "필터 없음"
    
    print(f"\n3. 얼굴 인식 테스트 - 필터: {filter_str}")
    
    data = {"image": image_data}
    if param1:
        data["param1"] = param1
    if param2:
        data["param2"] = param2
    
    response = requests.post(f"{BASE_URL}/api/face/recognize", json=data)
    print(f"상태: {response.status_code}")
    result = response.json()
    print(f"응답: {json.dumps(result, indent=2, ensure_ascii=False)}")
    
    if result.get('success') and result.get('matched'):
        print(f"✅ 인식 성공! member_id: {result.get('member_id')}, 유사도: {result.get('similarity')}")
    else:
        print("❌ 인식 실패")
    print("-" * 50)
    return result

def test_recognize_for_checkin(image_data, comp_cd, bcoff_cd):
    """체크인용 인식 테스트 (레거시 파라미터 → param1/param2 자동 변환)"""
    print(f"\n4. 체크인용 인식 테스트 - comp_cd: {comp_cd}, bcoff_cd: {bcoff_cd}")
    
    data = {
        "image": image_data,
        "comp_cd": comp_cd,
        "bcoff_cd": bcoff_cd,
        "security_level": 3
    }
    
    response = requests.post(f"{BASE_URL}/api/face/recognize_for_checkin", json=data)
    print(f"상태: {response.status_code}")
    result = response.json()
    print(f"응답: {json.dumps(result, indent=2, ensure_ascii=False)}")
    
    if result.get('success') and result.get('matched'):
        print(f"✅ 체크인 허용! member_id: {result.get('member_id')}")
    else:
        print("❌ 체크인 거부")
    print("-" * 50)
    return result

def run_all_tests():
    """모든 테스트 실행"""
    print("=" * 70)
    print("param1/param2 필터링 종합 테스트")
    print("=" * 70)
    
    # 1. 헬스 체크
    health = test_health_check()
    if health.get('status') != 'healthy':
        print("❌ 서버가 정상적이지 않습니다.")
        return
    
    # 테스트 이미지 준비 (실제 이미지 경로로 변경 필요)
    # image_data = load_test_image("test_face.jpg")
    
    # 테스트용 더미 이미지 데이터
    dummy_image = "data:image/jpeg;base64,/9j/4AAQSkZJRgABAQEASABIAAD/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAABAAEDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAf/xAAUEAEAAAAAAAAAAAAAAAAAAAAA/8QAFQEBAQAAAAAAAAAAAAAAAAAAAAX/xAAUEQEAAAAAAAAAAAAAAAAAAAAA/9oADAMBAAIRAxEAPwCdABmX/9k="
    
    print("\n🔹 시나리오 1: 다양한 지점에 회원 등록")
    # 회사 A, 지점 1
    test_register_with_params("MEM2025001", dummy_image, "COMP_A", "BRANCH_01")
    # 회사 A, 지점 2
    test_register_with_params("MEM2025002", dummy_image, "COMP_A", "BRANCH_02")
    # 회사 B, 지점 1
    test_register_with_params("MEM2025003", dummy_image, "COMP_B", "BRANCH_01")
    
    print("\n🔹 시나리오 2: 필터링 없이 전체 조회")
    test_recognize_with_filtering(dummy_image)
    
    print("\n🔹 시나리오 3: 특정 회사만 필터링")
    test_recognize_with_filtering(dummy_image, param1="COMP_A")
    
    print("\n🔹 시나리오 4: 특정 회사의 특정 지점만 필터링")
    test_recognize_with_filtering(dummy_image, param1="COMP_A", param2="BRANCH_01")
    
    print("\n🔹 시나리오 5: 레거시 파라미터 호환성 테스트")
    test_recognize_for_checkin(dummy_image, comp_cd="COMP_A", bcoff_cd="BRANCH_01")
    
    print("\n✅ 모든 테스트 완료!")

if __name__ == "__main__":
    run_all_tests()

File: face_recognition/examples/test_param_filtering.php
<?php
/**
 * param1/param2 필터링 테스트
 * 
 * 데이터베이스에 param1, param2 컬럼 추가 후 테스트
 */

require_once 'InsightFaceClient.php';

// 테스트 설정
$client = new InsightFaceClient('http://localhost:5002');

echo "======================================\n";
echo "param1/param2 필터링 PHP 테스트\n";
echo "======================================\n\n";

// 테스트용 이미지 (실제 이미지로 교체 필요)
$testImage = 'data:image/jpeg;base64,/9j/4AAQSkZJRgABAQEASABIAAD/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAABAAEDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAf/xAAUEAEAAAAAAAAAAAAAAAAAAAAA/8QAFQEBAQAAAAAAAAAAAAAAAAAAAAX/xAAUEQEAAAAAAAAAAAAAAAAAAAAA/9oADAMBAAIRAxEAPwCdABmX/9k=';

// 1. 헬스 체크
echo "1. 헬스 체크\n";
$health = $client->healthCheck();
print_r($health);
echo "\n----------------------------\n\n";

// 2. 회원 등록 테스트 (param1, param2 포함)
echo "2. 회원 등록 테스트\n";

// 회사 A, 지점 1
echo "- 회사 A, 지점 1에 회원 등록\n";
$result1 = $client->registerFace(
    'MEM2025001', 
    $testImage, 
    3, 
    '테스트 등록 - 회사A 지점1',
    'COMP_A',    // param1
    'BRANCH_01'  // param2
);
print_r($result1);

// 회사 A, 지점 2
echo "\n- 회사 A, 지점 2에 회원 등록\n";
$result2 = $client->registerFace(
    'MEM2025002', 
    $testImage, 
    3, 
    '테스트 등록 - 회사A 지점2',
    'COMP_A',    // param1
    'BRANCH_02'  // param2
);
print_r($result2);

// 회사 B, 지점 1
echo "\n- 회사 B, 지점 1에 회원 등록\n";
$result3 = $client->registerFace(
    'MEM2025003', 
    $testImage, 
    3, 
    '테스트 등록 - 회사B 지점1',
    'COMP_B',    // param1
    'BRANCH_01'  // param2
);
print_r($result3);

echo "\n----------------------------\n\n";

// 3. 얼굴 인식 테스트 (다양한 필터링)
echo "3. 얼굴 인식 테스트\n";

// 필터 없이 전체 조회
echo "- 필터 없이 전체 조회\n";
$recogAll = $client->recognizeFace($testImage);
printRecognitionResult($recogAll);

// 회사 A만 필터링
echo "\n- 회사 A만 필터링\n";
$recogCompA = $client->recognizeFace($testImage, 'COMP_A');
printRecognitionResult($recogCompA);

// 회사 A의 지점 1만 필터링
echo "\n- 회사 A의 지점 1만 필터링\n";
$recogCompABranch1 = $client->recognizeFace($testImage, 'COMP_A', 'BRANCH_01');
printRecognitionResult($recogCompABranch1);

// 회사 B의 지점 1만 필터링
echo "\n- 회사 B의 지점 1만 필터링\n";
$recogCompBBranch1 = $client->recognizeFace($testImage, 'COMP_B', 'BRANCH_01');
printRecognitionResult($recogCompBBranch1);

echo "\n----------------------------\n\n";

// 4. 레거시 파라미터 호환성 테스트
echo "4. 레거시 파라미터 호환성 테스트\n";

// 기존 방식으로 호출 (comp_cd, bcoff_cd 사용)
echo "- 기존 방식으로 체크인 인식 (comp_cd, bcoff_cd)\n";
$checkinResult = $client->recognizeFaceForCheckin(
    $testImage,
    'COMP_A',    // comp_cd → param1으로 자동 변환
    'BRANCH_01'  // bcoff_cd → param2로 자동 변환
);
printCheckinResult($checkinResult);

// 레거시 recognizeFace 호출
echo "\n- 레거시 방식으로 일반 인식 (comp_cd, bcoff_cd)\n";
$legacyRecog = $client->recognizeFace($testImage, null, null, 'COMP_A', 'BRANCH_01');
printRecognitionResult($legacyRecog);

echo "\n======================================\n";
echo "테스트 완료!\n";
echo "======================================\n";

// 헬퍼 함수들
function printRecognitionResult($result) {
    if ($result['success'] && $result['matched']) {
        echo "✅ 인식 성공! ";
        echo "member_id: " . $result['member_id'] . ", ";
        echo "유사도: " . $result['similarity'] . "\n";
    } else {
        echo "❌ 인식 실패\n";
    }
}

function printCheckinResult($result) {
    if ($result['success'] && $result['matched'] && $result['checkin_allowed']) {
        echo "✅ 체크인 허용! ";
        echo "member_id: " . $result['member_id'] . ", ";
        echo "유사도: " . $result['similarity'] . "\n";
    } else {
        echo "❌ 체크인 거부\n";
        if (isset($result['security_checks'])) {
            echo "보안 검사 결과: ";
            print_r($result['security_checks']);
        }
    }
}

// 실제 환경에서 사용하는 예제
echo "\n\n--- 실제 사용 예제 ---\n";
echo '
// 세션에서 지점 정보 가져오기
$param1 = $_SESSION["comp_cd"] ?? "DEFAULT_COMP";
$param2 = $_SESSION["bcoff_cd"] ?? "DEFAULT_BRANCH";

// 회원 등록
$result = $client->registerFace(
    $memberId,
    $imageData,
    3,
    "회원 등록",
    $param1,  // 회사 코드
    $param2   // 지점 코드
);

// 얼굴 인식 (해당 지점만)
$result = $client->recognizeFace(
    $imageData,
    $param1,  // 해당 회사만
    $param2   // 해당 지점만
);
';
?>